core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckStrategy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint.healthcheck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.linecorp.armeria.client.Endpoint;
+
+public interface HealthCheckStrategy {
+    List<Endpoint> NOT_CHANGE = new ArrayList<>();
+
+    void updateCandidate(List<Endpoint> candidates);
+
+    List<Endpoint> getCandidates();
+
+    default void updateHealth(Endpoint endpoint, double health) {
+        // NOP
+    }
```

Comment:
I guess we don't have to make this a default method because most implementation have different implementations? No-op implementation probably only applied static implementations.

Reply:
Fixed